pyk/src/pyk/kore/syntax.py

```diff
@@ -533,6 +533,167 @@ def write(self, output: IO[str]) -> None:
+
+    def let(
+        self,
+        *,
+        symbol: str | SymbolId | None = None,
+        sorts: Iterable | None = None,
+        args: Iterable | None = None,
+    ) -> LeftAssoc:
+        symbol = symbol if symbol is not None else self.symbol
+        sorts = sorts if sorts is not None else self.sorts
+        args = args if args is not None else self.args
+        return LeftAssoc(symbol=symbol, sorts=sorts, args=args)
+
+    def let_patterns(self, patterns: Iterable[Pattern]) -> LeftAssoc:
+        return self.let(args=patterns)
+
+    @property
+    def pattern(self) -> Pattern:
+        res = self.args[0]
+        for arg in self.args[1:]:
+            res = App(self.symbol, self.sorts, (res, arg))
+        return res
+
+    @classmethod
+    def _tag(cls) -> str:
+        return 'LeftAssoc'
+
+    @classmethod
+    def kore_symbol(cls) -> str:
+        return '\\left-assoc'
+
+    @classmethod
+    def _from_dict(cls: type[LeftAssoc], dct: Mapping[str, Any], patterns: list[Pattern]) -> LeftAssoc:
+        return LeftAssoc(
+            symbol=dct['symbol'],
+            sorts=tuple(Sort.from_dict(sort) for sort in dct['sorts']),
+            args=patterns,
+        )
+
+
+@final
+@dataclass(frozen=True)
+class RightAssoc(Assoc):
+    symbol: str
+    sorts: tuple[Sort, ...]
+    args: tuple[Pattern, ...]
+
+    def __init__(self, symbol: str | SymbolId, sorts: Iterable[Sort] = (), args: Iterable[Pattern] = ()):
+        if isinstance(symbol, str):
+            symbol = SymbolId(symbol)
+
+        args = tuple(args)
+        if not args:
+            raise ValueError("Expected non-empty iterable for 'args'")
+
+        object.__setattr__(self, 'symbol', symbol.value)
+        object.__setattr__(self, 'sorts', tuple(sorts))
+        object.__setattr__(self, 'args', args)
+
+    def let(
+        self,
+        *,
+        symbol: str | SymbolId | None = None,
+        sorts: Iterable | None = None,
+        args: Iterable | None = None,
+    ) -> RightAssoc:
+        symbol = symbol if symbol is not None else self.symbol
+        sorts = sorts if sorts is not None else self.sorts
+        args = args if args is not None else self.args
+        return RightAssoc(symbol=symbol, sorts=sorts, args=args)
+
+    def let_patterns(self, patterns: Iterable[Pattern]) -> RightAssoc:
+        return self.let(args=patterns)
+
+    @property
+    def pattern(self) -> Pattern:
+        res = self.args[-1]
+        for arg in reversed(self.args[:-1]):
+            res = App(self.symbol, (), (arg, res))
+        return res
+
+    @classmethod
+    def _tag(cls) -> str:
+        return 'RightAssoc'
+
+    @classmethod
+    def kore_symbol(cls) -> str:
+        return '\\right-assoc'
+
+    @classmethod
+    def of(
+        cls: type[RightAssoc],
+        symbol: str,
+        sorts: Iterable[Sort] = (),
+        patterns: Iterable[Pattern] = (),
+    ) -> RightAssoc:
+        return RightAssoc(symbol=symbol, sorts=sorts, args=patterns)
```

Comment:
Is this meant to be removed? It isn't in `LeftAssoc`.

Reply:
Nice catch!